stablehlo/dialect/TypeInference.cpp

```diff
@@ -2484,12 +2484,19 @@ LogicalResult inferReduceWindowOp(
     return failure();
 
   for (size_t i = 0; i < inputArgTypes.size(); ++i) {
-    if (!inputArgTypes[i].hasRank())
+    if (!inputArgTypes[i].hasRank()) {
       inferredReturnShapes.emplace_back(inputArgTypes[i].getElementType());
-    else
-      inferredReturnShapes.emplace_back(
-          inferWindowOutputShape(inputArgTypes[i].getShape(), inferredWindow),
-          inputArgTypes[i].getElementType());
+    } else {
+      auto inputRankedType = inputs[i].getType().cast<RankedTensorType>();
```

Comment:
nit: dyn_cast followed by (!inputRankedType) instead of hasRank followed by cast.